### PR TITLE
Check if the window passed to _xdo_send_key has focus, and use XTestF…

### PR DESCRIPTION
**Pull request by mpnolan**
…akeKeyEvent if so.

This allows XTestFakeKeyEvent, rather than XSendEvent, to be used in a command chain like this:
xdotool search --class App windowfocus --sync key a

### CODE REVIEW

**Comment by jordansissel** [comment]
I think I understand your proposal and your patch (Thank you, by the way!!). I think it's worth noting that doing `key --window 0 a` will send 'a' to the current window (There's a special window value xdo uses for knowing when to act on the "current window" and that is 0).

Given that, I like your proposal very much. It makes a lot of sense to have `key` use XTestFakeKeyEvent if the window on the stack is the current window. ++

**Comment by phil294** [comment]
this was a good change, but I think something's not completely right. Something up front: This is probably an edge case and may not affect anyone else really, but I think it's worth noting.

The following variations need to be done with an instance of `Mousepad` open. Apparently, Mousepad is an application that somehow eats ignores XSend events but XTest always works. 
```
echo "Focus Mousepad *manually* in the next 5 seconds"
sleep 5
xdotool key a # Always works, because a missing window id means XTest
```
```
# Sets focus using XSetInputFocus https://github.com/jordansissel/xdotool/blob/ee38f704eee6ef0a69eba9b16c8477f87b5f31bf/xdo.c#L413
xdotool search --onlyvisible --name Mousepad windowfocus
xdotool search --onlyvisible --name Mousepad key a # Works!
```
```
xdotool search --onlyvisible --name Mousepad windowfocus key a # Also works, this is a shorthand of the two above commands
```
```
echo "Focus Mousepad *manually* in the next 5 seconds"
sleep 5
xdotool search --onlyvisible --name Mousepad key a # Does not work! Nothing is sent.
```
So I don't know why, but manually focussing the window does not seem to properly set the focus flag? Or maybe it does, but it's not picked up by xdotool? You can verify with `xdotool getwindowfocus` or `xdotool getactivewindow` that it's actually the exact same window ID all the time, no matter what you do, but the detection `if (focuswin == window) {` from this PR only seems to work when the focus was set programmatically.

**TL;DR**: `xdotool --onlyvisible --name Mousepad key a` never works.
